src/jdk.jdwp.agent/share/native/libjdwp/util.c

```diff
@@ -1977,21 +1977,29 @@ eventIndexInit(void)
 }
 
 jdwpEvent
-eventIndex2jdwp(EventIndex i)
+eventIndex2jdwp(EventIndex ei)
 {
-    if ( i < EI_min || i > EI_max ) {
-        EXIT_ERROR(AGENT_ERROR_INVALID_INDEX,"bad EventIndex");
+    jdwpEvent event = 0;
+    if (ei >= EI_min || ei >= EI_max) {
+        event = index2jdwp[ei - EI_min];
     }
-    return index2jdwp[i-EI_min];
+    if (event == 0) {
+        EXIT_ERROR(AGENT_ERROR_INVALID_INDEX, "bad EventIndex");
+    }
+    return event;
 }
 
 jvmtiEvent
-eventIndex2jvmti(EventIndex i)
+eventIndex2jvmti(EventIndex ei)
 {
-    if ( i < EI_min || i > EI_max ) {
-        EXIT_ERROR(AGENT_ERROR_INVALID_INDEX,"bad EventIndex");
+    jvmtiEvent event = 0;
+    if (ei >= EI_min || ei >= EI_max) {
```

Comment:
Should be "(ei >= EI_min && ei <= EI_max"

Reply:
Fixed.